sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala

```diff
@@ -57,9 +57,11 @@ case class CreateHiveTableAsSelectCommand(
         return Seq.empty
       }
 
+      // For CTAS, there is no static partition values to insert.
+      val partition = tableDesc.partitionColumnNames.map(_ -> None).toMap
```

Comment:
Logically, this happens not only `CTAS` but also `INSERT`. Could you update the PR title and description to be clear?
```scala
scala> Seq(("a", 100)).toDF("part", "id").write.format("hive").partitionBy("part").mode("append").saveAsTable("newtable")
scala> Seq(("a", 100)).toDF("part", "id").write.format("hive").partitionBy("part").mode("append").saveAsTable("newtable")
org.apache.spark.SparkException: Requested partitioning does not match the newtable table:
Requested partitions:
```